test/integration/edit_bulk_assignment_weight_test.rb

```diff
@@ -0,0 +1,56 @@
+require 'test_helper'
+
+class EditBulkAssignmentWeightTest < ActionDispatch::IntegrationTest
+  include Devise::Test::IntegrationHelpers
+  setup do
+    Capybara.default_driver = :selenium_chrome_headless
+    make_standard_course
```

Comment:
These tests are good but not quite sufficient for scenario 1: once exam assignments exist, they'll be part of the weights page, even if you don't change their weights specifically. So you'll need something besides `make_standard_course`